File: 2024.1/lab1/python/createFiles.py
import os
import random
import string
import sys
import shutil

def create_random_string(length):
    """Cria uma string aleatória de comprimento especificado."""
    letters = string.ascii_letters + string.digits
    return ''.join(random.choice(letters) for _ in range(length))

def create_directories_and_files(root, N):
    """Cria o diretório root e N arquivos em cada diretório."""
    os.makedirs(root, exist_ok=True)
    
    for j in range(1, N + 1):
        file_path = os.path.join(root, f'file_{j}.txt')
        with open(file_path, 'w') as f:
            random_string = create_random_string(100)  # Pode ajustar o comprimento da string
            f.write(random_string)

def clean_directories_and_files(root):
    """Remove recursivamente o diretório root e todo o seu conteúdo."""
    try:
        shutil.rmtree(root)
        print(f"Conteúdo de '{root}' removido com sucesso.")
    except OSError as e:
        print(f"Erro ao remover conteúdo de '{root}': {e}")

N = int(sys.argv[1])

# Exemplo de uso
root_directory = 'root_directory'
clean_directories_and_files(root_directory)
create_directories_and_files(root_directory, N)

print(f"Diretórios e arquivos criados em '{root_directory}'")

File: 2024.1/lab1/java/CreateFiles.java
import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.util.Random;

public class CreateFiles {

    private static final String CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
    private static final Random RANDOM = new Random();

    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Informe o total de arquivos");
            System.exit(1);
        }

        int N = Integer.parseInt(args[0]);
        String rootDirectory = "../root_directory";

        try {
            cleanDirectoriesAndFiles(rootDirectory);
            createDirectoriesAndFiles(rootDirectory, N);
            System.out.println("Diretórios e arquivos criados em '" + rootDirectory + "'");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    private static String createRandomString(int length) {
        StringBuilder sb = new StringBuilder(length);
        for (int i = 0; i < length; i++) {
            sb.append(CHARACTERS.charAt(RANDOM.nextInt(CHARACTERS.length())));
        }
        return sb.toString();
    }

    private static void createDirectoriesAndFiles(String root, int N) throws IOException {
        File rootDir = new File(root);
        if (!rootDir.exists()) {
            if (!rootDir.mkdirs()) {
                throw new IOException("Falha ao criar o diretório: " + root);
            }
        }

        for (int j = 1; j <= N; j++) {
            File file = new File(root, "file_" + j + ".txt");
            try (FileWriter writer = new FileWriter(file)) {
                String randomString = createRandomString(100);  // Pode ajustar o comprimento da string
                writer.write(randomString);
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }

    private static void cleanDirectoriesAndFiles(String root) throws IOException {
        File rootDir = new File(root);
        if (rootDir.exists()) {
            deleteRecursive(rootDir);
            System.out.println("Conteúdo de '" + root + "' removido com sucesso.");
        }
    }

    private static void deleteRecursive(File fileOrDir) {
        if (fileOrDir.isDirectory()) {
            for (File child : fileOrDir.listFiles()) {
                deleteRecursive(child);
            }
        }
        fileOrDir.delete();
    }
}
